inception-workload-dynamic/src/main/java/de/tudarmstadt/ukp/inception/workload/dynamic/annotation/DynamicAnnotatorWorkflowActionBarItemGroup.java

```diff
@@ -87,50 +107,99 @@ public String getStateClass()
+            // No annotation documents in the state INPROGRESS, now select a new one
+            // depending on the workload strategy selected
+            WorkloadManager currentWorkload = workloadManagementService
+                    .loadOrCreateWorkloadManagerConfiguration(project);
+
+            // Get all documents for which the state is NEW, or which have not been created yet.
+            List<SourceDocument> sourceDocuments = workloadManagementService
+                    .getAnnotationDocumentListForUser(project, user);
+
+            // Switch for all workflow types which are available. If a new one is created.
+            // simply add a new case here.
+            switch (dynamicWorkloadExtension.readTraits(currentWorkload).getWorkflowType()) {
+
+            // Go through all documents in a random order
+            case (RANDOMIZED_WORKFLOW):
+                // Shuffle the List then call loadNewDocument
+                Collections.shuffle(sourceDocuments);
+                loadNewDocument(sourceDocuments, project, currentWorkload, getAnnotationPage(),
+                        _target);
+                break;
+
+            // Default workflow selected, nothing to change in the list
+            default:
+                loadNewDocument(sourceDocuments, project, currentWorkload, getAnnotationPage(),
+                        _target);
+                break;
             }
         });
         finishDocumentDialog.show(aTarget);
     }
+
+    private void loadNewDocument(List<SourceDocument> aSourceDocuments, Project aProject,
+            WorkloadManager aCurrentWorkload, AnnotationPageBase aPage, AjaxRequestTarget aTarget)
+    {
+        // Go through all documents of the list
+        for (SourceDocument doc : aSourceDocuments) {
+            // Check if there are less annotators working on the selected document than
+            // the default number of annotation set by the project manager
+            if ((workloadManagementService.getAmountOfUsersWorkingOnADocument(doc, aProject)
+                    + 1) <= (dynamicWorkloadExtension.readTraits(aCurrentWorkload)
+                            .getDefaultNumberOfAnnotations())) {
+                // This was the case, so load the document and return
+                aPage.getModelObject().setDocument(doc,
+                        documentService.listSourceDocuments(aProject));
+                aPage.actionLoadDocument(aTarget);
+                return;
+            }
+        }
+        // No documents left, return to homepage and show corresponding message
+        redirectUSerToHomePage();
+
+    }
+
+    private void redirectUSerToHomePage()
+    {
+        // Nothing left, so returning to homepage and showing hint
+        getAnnotationPage().setResponsePage(getAnnotationPage().getApplication().getHomePage());
+        getAnnotationPage().getSession().info(
```

Comment:
I'd set the info before the redirect. Might not make a difference in practice though.